File: TypePonto/Models/Marcacao.cs
using System;

namespace TypePonto.Models
{
    public class Marcacao
    {
        public int Id { get; set; }
        public int FuncionarioId { get; set; }
        public Funcionario funcionario { get; set; }

        public DateTime DataMarcacao { get; set; }

        public DateTime HoraEntrada { get; set; }

        public DateTime HoraSaidaAlmoco { get; set; }

        public DateTime HoraVoltaAlmoco { get; set; }

        public DateTime Saida { get; set; }


    }
}

File: FRONT/src/app/models/marcacao.ts
export interface Marcacao
{
    id?: number,
    funcionarioId: Number
    dataMarcacao: string;
    funcionario?: string;
    horaEntrada: string;
    horaSaidaAlmoco: string;
    horaVoltaAlmoco: string;
    saida: string;
}